lib/Echidna/Types/Config.hs

```diff
@@ -64,9 +64,11 @@ data Env = Env
   { cfg :: EConfig
   , dapp :: DappInfo
 
-  -- | Shared between all workers. Events are fairly rare so contention is
-  -- minimal.
-  , eventQueue :: Chan (Int, LocalTime, CampaignEvent)
+  , eventHandlers :: IORef [(Int, LocalTime, CampaignEvent) -> IO ()]
+
+  -- mainly for handling events, but can be used for any purpose
+  -- `wait` is called on this group before echidna closes
+  , threadGroup :: ThreadGroup
```

Comment:
We should move this to `spawnListener` and make all the handlers spawned async

Reply:
Not sure I follow? I got rid of `spawnListener`, and all handlers are spawned async now